language-plutus-core/test/Spec.hs

```diff
@@ -109,16 +111,35 @@ asGolden f file = goldenVsString file (file ++ ".golden") (asIO f file)
 testsType :: [FilePath] -> TestTree
 testsType = testGroup "golden type synthesis tests" . fmap (asGolden printType)
 
+testsNormalizeType :: [FilePath] -> TestTree
+testsNormalizeType = testGroup "golden type synthesis + normalization tests" . fmap (asGolden (printNormalizeType True))
+
 testsGolden :: [FilePath] -> TestTree
 testsGolden = testGroup "golden tests" . fmap (asGolden (format defPrettyConfigPlcClassic))
 
 testsRewrite :: [FilePath] -> TestTree
 testsRewrite = testGroup "golden rewrite tests" . fmap (asGolden (format debugPrettyConfigPlcClassic))
 
+appAppLamLam :: MonadQuote m => m (Type TyNameWithKind ())
+appAppLamLam = do
+    x <- liftQuote (TyNameWithKind <$> freshTyName ((), Type ()) "x")
+    y <- liftQuote (TyNameWithKind <$> freshTyName ((), Type ()) "y")
+    pure $
+        TyApp ()
+            (TyApp ()
+                 (TyLam () x (Type ()) (TyLam () y (Type ()) $ TyVar () y))
+                 (TyBuiltin () TyInteger))
+            (TyBuiltin () TyInteger)
+
+testLam :: Either (TypeError ()) String
+testLam = fmap (prettyStringBy defPrettyConfigPlcClassic) . runQuote . runExceptT $ runTypeCheckM 100 False $
```

Comment:
There is `prettyPlcDefString` in `Language.PlutusCore.Pretty` (which I stupidly forgot to comment).

Reply:
Ah wonderful, thanks.